### DISCUSSION TITLE
Site needs some desperate improvements in loading time

### DISCUSSION THREAD

**Discussion by Andre601** [simple-icons/simple-icons#7125] [General]
### Issue

Whenever I visit https://simpleicons.org/ do I have to wait several seconds, if not even minutes to have a usable website.
Despite the tab indicating that the loading of the page has finished, does the page not run smoothly for quite a while. Probably because it loads all the SVG icons before doing anything else.

This is an extremely annoying and unpleasant experience that imo should be avoided.

### Ideas

I'm unfortunately not a web dev and don't have the front- or backend knowledge to suggest any proper improvements for the site.
I tried using [web.dev](https://web.dev) to analyze the site and get suggestions on how it could be improved, but the site encountered an error when analyzing the site... twice! First time just an "Error" and second time a "TypeError: Failed to fetch", so no suggestions from there.

The only thing I could think of is, to improve the SVG loading. Be it by having them Lazy load (if not done already) or only load the "visible" ones (Load any other that is exposed through scrolling when they are actually shown)?

Another idea could be pagination (Maybe by alphabet?) to not have **over 2k SVGs loaded at the same time**?

### Context and personal setup

To counter any argument that this could be because of my setup (Be a "me"-problem) I can give you the following stats:

- Running Windows 10 Pro version
- Running Microsoft Edge v97.0.1072.76 (Official Build, 64-Bit)
- Results from Speedtest.net ([Proof](https://www.speedtest.net/result/12708566435)):
  - Ping: 7ms
  - Download: 99.24 Mbps
  - Upload: 103.21 Mbps

### Final words

I hope there will be some improvements.

While people can come with the argument "Use a different browser" do I feel like that this would only push the issue under the rug. It should be a goal for a website to support multiple browsers, if they are maintained and updated and not just suggest some other browser to use.

I can also recall not having those issues with the old version of the website, which makes the redesign a little worse, since I like the general look and feel of it.

**Comment by sachinraja** [comment]
I agree and we are aware of this. Please create an issue at [simple-icons-website](https://github.com/simple-icons/simple-icons-website). We are currently lazy loading the SVGs (you can verify this). I'm not sure if storing all the icons in memory or using a ton of `IntersectionObserver`s is a good idea, but I have also looked into loading only the visible icons and that would probably be the best way to fix the performance problems.

**Comment by adsingh14** [comment]
https://simpleicons.org website crash when you click/tap on search bar while loading on both Chromium and FF browsers.

**Comment by LitoMore** [comment]
We plan to use React to rewrite our website. See https://github.com/simple-icons/simple-icons-website/issues/255.

**Comment by mondeja** [comment]
Hi guys,

After some PRs today seems that the main problem has been solved. I'll greatly appreciate some feedback if you still experimenting these types of performance problems or sharing your thoughs if you have more ideas on https://github.com/simple-icons/simple-icons-website/issues/284

Cheers and thanks in advance

**Comment by adamrusted** [comment]
Closing out, as it would appear the issues surfaced at the time have been dealt with. For any queries/issues regarding the website, please be sure to post them at https://github.com/simple-icons/simple-icons-website
